enclave/core/switchlesscalls.c

```diff
@@ -65,15 +83,30 @@ oe_result_t oe_handle_init_switchless(uint64_t arg_in)
         OE_RAISE(OE_INVALID_PARAMETER);
     }
 
+    // Check if already initialized. Doing this check here means that
+    // simultaneous calls to oe_handle_init_switchless may duplicate the above
+    // checks, however this is an acceptably small overhead to ensure that at
+    // least one initialization succeeds in a thread-safe way.
+    if (!oe_atomic_compare_and_swap(
```

Comment:
The initialized flag must be set *after* initialization has been done; otherwise another thread could incorrectly think that initialization has been done. This flag has to be set after line 103.

Reply:
Fixed